core/src/main/resources/reference.conf

```diff
@@ -1785,14 +1785,41 @@ datastax-java-driver {
       # Overridable in a profile: no
       enabled = true
 
-      # The list of keyspaces for which schema and token metadata should be maintained. If this
-      # property is absent or empty, all existing keyspaces are processed.
+      # The keyspaces for which schema and token metadata should be maintained.
       #
-      # Required: no
+      # Each element can be one of the following:
+      # 1. An exact name inclusion, for example "Ks1". If the name is case-sensitive, it must appear
+      #    in its exact case.
+      # 2. An exact name exclusion, for example "!Ks1".
+      # 3. A regex inclusion, enclosed in slashes, for example "/^Ks.*/". The part between the
+      #    slashes must follow the syntax rules of java.util.regex.Pattern.
+      # 4. A regex exclusion, for example "!/^Ks.*/".
+      #
+      # If the list is empty, or the option is unset, all keyspaces will match. Otherwise:
+      #
+      # If a keyspace matches an exact name inclusion, it is always included, regardless of what any
+      # other rule says.
+      # Otherwise, if it matches an exact name exclusion, it is always excluded, regardless of what
+      # any regex rule says.
+      # Otherwise, if there are regex rules:
+      # - if they're only inclusions, the keyspace must match at least one of them.
+      # - if they're only exclusions, the keyspace must match none of them.
+      # - if they're both, the keyspace must match at least one inclusion and none of the
+      #   exclusions.
+      #
+      # If an element is malformed, or if its regex has a syntax error, a warning is logged and that
+      # single element is ignored.
+      #
+      # Try to use only exact name inclusions if possible. This allows the driver to filter on the
+      # server side with a WHERE IN clause. If you use any other rule, it has to fetch all system
+      # rows and filter on the client side.
+      #
+      # Required: no. The default value excludes all Cassandra and DSE system keyspaces. If the
+      #   option is unset, this is interpreted as "include all keyspaces".
       # Modifiable at runtime: yes, the new value will be used for refreshes issued after the
       #   change.
       # Overridable in a profile: no
-      // refreshed-keyspaces = [ "ks1", "ks2" ]
+      refreshed-keyspaces = [ "!system", "!/^system_.*/", "!/^dse_.*/", "!solr_admin" ]
```

Comment:
So by default we are not using server-side filtering. I wonder how many users will care about changing the defaults to benefit from server-side filtering? Which brings the question: will server-side filtering be used at all?

Reply:
> So by default we are not using server-side filtering

Correct. This is still an improvement, 4.9 fetches and parses everything, 4.10 will fetch everything but only parse a subset.

I think we can still expect that filtering to one or two specific keyspaces will be a common occurrence. Adding a WHERE clause is easy and brings a good improvement.